src/docfx/template/XrefTagParser.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class XrefTagParser
+    {
+        /// <summary>
+        /// Using `href` property to indicate xref spec resolve success.
+        /// </summary>
+        private const string XrefTagTemplate =
+        "{{#href}}" +
+        "  @resolvedTag" +
+        "{{/href}}" +
+        "{{^href}}" +
+        "  <span> {{.}} </span>" +
+        "{{/href}}";
+
+        private static readonly char[] s_trimChars = new[] { '{', ' ', '}' };
+
+        private static readonly Regex s_xrefTagMatcher = new Regex(@"<xref(.*?)\/>", RegexOptions.IgnoreCase);
+
+        public static string ProcessXrefTag(string templateStr) => s_xrefTagMatcher.Replace(templateStr, ReplaceXrefTag);
+
+        private static string ReplaceXrefTag(Match match)
+        {
+            var xrefTag = HtmlUtility.LoadHtml(match.Value).ChildNodes.FindFirst("xref");
+            var uidName = default(string);
+            var partialName = default(string);
+            foreach (var attribute in xrefTag.Attributes)
+            {
+                if (string.Equals(attribute.Name, "uid", StringComparison.OrdinalIgnoreCase))
+                    uidName = attribute.Value.Trim(s_trimChars);
+                else if (string.Equals(attribute.Name, "template", StringComparison.OrdinalIgnoreCase))
+                    partialName = attribute.Value.Trim(s_trimChars);
+            }
+
+            var resolvedTag = partialName == null
+                ? "<a href=\"{{href}}\"> {{uid}} </a>"
+                : "{{> " + partialName + "}}";
+            var resultTemplate = XrefTagTemplate.Replace("@resolvedTag", resolvedTag);
+
+            if (uidName == ".")
```

Comment:
what if `uidName` is null?

Reply:
Nice catch !
Currently there's no such case, we can simply throw here to avoid it.

There's however one exception for `ProfileList`
```html
<xref href="{{uid}}" title="{{name}}">
  {{#image}}
    <img class="profile-list-image" src="{{src}}" alt="">
  {{/image}}
</xref>
```
Current regex won't match it, so it's safe here. We need to add a TODO for `ProfileList` thougth.